debezium-connector-sqlserver/src/test/java/io/debezium/connector/sqlserver/TablesWithUniqueIndexOnlyIT.java

```diff
@@ -0,0 +1,129 @@
+        Assertions.assertThat(records.recordsForTopic("server1.dbo.t1").get(0).keySchema().field("key1")).isNotNull();
+        Assertions.assertThat(records.recordsForTopic("server1.dbo.t1").get(0).keySchema().field("key2")).isNotNull();
+    }
+
+    @Test
+    public void shouldProcessFromStreaming() throws Exception {
+        connection = TestHelper.testConnection();
+        connection.execute(DDL_STATEMENTS + DML_STATEMENTS);
+
+        TestHelper.enableTableCdc(connection, "t1", "t1_CT", Collect.arrayListOf("key1", "key2", "data"));
+
+        start(SqlServerConnector.class, TestHelper.defaultConfig()
+                .with(SqlServerConnectorConfig.SNAPSHOT_MODE, SnapshotMode.INITIAL)
+                .with(SqlServerConnectorConfig.MSG_KEY_COLUMNS, "dbo.t1:key1,key2;dbo.t2:key1,key2")
+                .build());
+        assertConnectorIsRunning();
+
+        final int expectedRecordsCount = 1;
+
+        SourceRecords records = consumeRecordsByTopic(expectedRecordsCount);
+        Assertions.assertThat(records.recordsForTopic("server1.dbo.t1").get(0).keySchema().field("key1")).isNotNull();
+        Assertions.assertThat(records.recordsForTopic("server1.dbo.t1").get(0).keySchema().field("key2")).isNotNull();
+
+        connection.execute("INSERT INTO t1 VALUES (2, 20, 'data2', 200);");
+
+        records = consumeRecordsByTopic(expectedRecordsCount);
+        Assertions.assertThat(records.recordsForTopic("server1.dbo.t1").get(0).keySchema().field("key1")).isNotNull();
+        Assertions.assertThat(records.recordsForTopic("server1.dbo.t1").get(0).keySchema().field("key2")).isNotNull();
+
+        connection.execute(DDL_STATEMENTS_STREAM);
+        TestHelper.enableTableCdc(connection, "t2", "t2_CT", Collect.arrayListOf("key1", "key2"));
+        waitForEnabledCdc(connection, "t2");
+        connection.execute("INSERT INTO t2 VALUES (2, 20, 'data2', 200);");
+
+        records = consumeRecordsByTopic(expectedRecordsCount);
+        Assertions.assertThat(records.recordsForTopic("server1.dbo.t2").get(0).keySchema().field("key1")).isNotNull();
+        Assertions.assertThat(records.recordsForTopic("server1.dbo.t2").get(0).keySchema().field("key2")).isNotNull();
+    }
+
+    private void waitForEnabledCdc(SqlServerConnection connection, String table) throws SQLException, InterruptedException {
+        Awaitility
+                .await("CDC " + table)
+                .atMost(1, TimeUnit.MINUTES)
+                .pollInterval(100, TimeUnit.MILLISECONDS)
+                .until(() -> TestHelper.isCdcEnabled(connection, table));
+    }
+
+    private void waitForDisabledCdc(SqlServerConnection connection, String table) throws SQLException, InterruptedException {
+        Awaitility
+                .await("CDC " + table)
+                .atMost(1, TimeUnit.MINUTES)
+                .pollInterval(100, TimeUnit.MILLISECONDS)
+                .until(() -> !TestHelper.isCdcEnabled(connection, table));
+    }
```

Comment:
We have those in `TablesWithoutPrimaryKeyIT`already; wanna move them to `TestHelper`?

Reply:
@gunnarmorling Yes, I plnned to do it but forgot at the end. Thaks for reminding me.